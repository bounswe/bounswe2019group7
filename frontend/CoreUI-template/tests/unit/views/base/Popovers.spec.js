import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Popovers from '@/views/base/Popovers'

Vue.use(BootstrapVue)

describe('Popovers.vue', () => {
  it('has a name', () => {
    expect(Popovers.name).toMatch('popovers')
  })
  it('has a created hook', () => {
    expect(typeof Popovers.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Popovers.data).toMatch('function')
    const defaultData = Popovers.data()
    expect(defaultData.show).toBe(false)
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Popovers)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Popovers', () => {
    const wrapper = shallowMount(Popovers)
    expect(wrapper.is(Popovers)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Popovers)
    expect(wrapper.find('header.card-header > div > strong').text()).toMatch('Bootstrap Popovers')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Popovers)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should have methods', () => {
    const wrapper = mount(Popovers)
    expect(typeof Popovers.methods.onOpen).toEqual('function')
    expect(wrapper.vm.onOpen()).toBeUndefined()
    expect(typeof Popovers.methods.onClose).toEqual('function')
    expect(wrapper.vm.onClose()).toBeUndefined()
    expect(typeof Popovers.methods.onEnable).toEqual('function')
    expect(wrapper.vm.onEnable()).toBeUndefined()
    expect(typeof Popovers.methods.onDisable).toEqual('function')
    expect(wrapper.vm.onDisable()).toBeUndefined()
  })
})
