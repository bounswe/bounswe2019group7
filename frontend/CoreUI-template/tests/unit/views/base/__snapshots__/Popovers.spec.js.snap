// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popovers.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Bootstrap Popovers
        </strong>
         
        <div
          class="card-header-actions"
        >
          <a
            class="card-header-action"
            href="https://bootstrap-vue.js.org/docs/components/popovers"
            rel="noreferrer noopener"
            target="_blank"
          >
            <small
              class="text-muted"
            >
              docs
            </small>
          </a>
        </div>
      </div>
       
      <b-row-stub
        tag="div"
      >
        <b-col-stub
          cols="6"
          tag="div"
        >
          <div
            class="my-3 text-center"
          >
            <b-btn-stub
              event="click"
              routertag="a"
              tag="button"
              target="_self"
              title="Popover Title"
              type="button"
              variant="secondary"
            >
              
              Hover Me
            
            </b-btn-stub>
          </div>
        </b-col-stub>
         
        <b-col-stub
          cols="6"
          tag="div"
        >
          <div
            class="my-3 text-center"
          >
            <b-btn-stub
              event="click"
              id="popoverButton-open"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              Button
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content=""
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="right"
              show="true"
              target="popoverButton-open"
              title="Popover"
              triggers="click"
            >
              
              I start 
              <strong>
                open
              </strong>
            </b-popover-stub>
          </div>
        </b-col-stub>
      </b-row-stub>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Popovers 
        </strong>
         
        <small>
          placement
        </small>
      </div>
       
      <div
        class="my-3"
      >
        <b-row-stub
          tag="div"
        >
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-topright"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              topright
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement topright"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="topright"
              target="exPopover1-topright"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-top"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              top
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement top"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="top"
              target="exPopover1-top"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-topleft"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              topleft
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement topleft"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="topleft"
              target="exPopover1-topleft"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-bottomright"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              bottomright
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement bottomright"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="bottomright"
              target="exPopover1-bottomright"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-bottom"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              bottom
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement bottom"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="bottom"
              target="exPopover1-bottom"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-bottomleft"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              bottomleft
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement bottomleft"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="bottomleft"
              target="exPopover1-bottomleft"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-righttop"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              righttop
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement righttop"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="righttop"
              target="exPopover1-righttop"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-right"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              right
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement right"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="right"
              target="exPopover1-right"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-lefttop"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              lefttop
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement lefttop"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="lefttop"
              target="exPopover1-lefttop"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-rightbottom"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              rightbottom
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement rightbottom"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="rightbottom"
              target="exPopover1-rightbottom"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-left"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              left
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement left"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="left"
              target="exPopover1-left"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
          <b-col-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <b-btn-stub
              event="click"
              id="exPopover1-leftbottom"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="primary"
            >
              
              leftbottom
            
            </b-btn-stub>
             
            <b-popover-stub
              boundary="scrollParent"
              boundarypadding="5"
              content="Placement leftbottom"
              delay="0"
              fallbackplacement="flip"
              offset="0"
              placement="leftbottom"
              target="exPopover1-leftbottom"
              title="Popover!"
              triggers="hover focus"
            />
          </b-col-stub>
        </b-row-stub>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      class="d-sm-down-none"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Popovers 
        </strong>
         
        <small>
          via properties or slots
        </small>
      </div>
       
      <b-row-stub
        tag="div"
      >
        <b-col-stub
          class="py-4 text-center"
          md="6"
          tag="div"
        >
          <b-btn-stub
            event="click"
            id="exPopover2"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="primary"
          >
            Using properties
          </b-btn-stub>
           
          <b-popover-stub
            boundary="scrollParent"
            boundarypadding="5"
            content="Embedding content using properties is easy"
            delay="0"
            fallbackplacement="flip"
            offset="0"
            placement="right"
            target="exPopover2"
            title="Prop Examples"
            triggers="hover focus"
          />
        </b-col-stub>
         
        <b-col-stub
          class="py-4 text-center"
          md="6"
          tag="div"
        >
          <b-btn-stub
            event="click"
            id="exPopover3"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="primary"
          >
            Using slots
          </b-btn-stub>
           
          <b-popover-stub
            boundary="scrollParent"
            boundarypadding="5"
            content=""
            delay="0"
            fallbackplacement="flip"
            offset="0"
            placement="right"
            target="exPopover3"
            title=""
            triggers="hover focus"
          >
            <template>
              Content via Slots
            </template>
            
            Embedding content 
            <span
              class="text-danger"
            >
              using slots
            </span>
            
            affords you 
            <em>
              greater 
              <strong>
                control.
              </strong>
            </em>
             and
            basic HTML support.
          
          </b-popover-stub>
        </b-col-stub>
      </b-row-stub>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Popovers 
        </strong>
         
        <small>
          show
        </small>
      </div>
       
      <div
        class="d-flex flex-column text-md-center"
      >
        <div
          class="p-2"
        >
          <b-btn-stub
            event="click"
            id="popoverButton-sync"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="primary"
          >
            I have a popover
          </b-btn-stub>
        </div>
         
        <div
          class="p-2"
        >
          <b-btn-stub
            class="px-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Toggle Popover
          </b-btn-stub>
           
          <b-popover-stub
            boundary="scrollParent"
            boundarypadding="5"
            content=""
            delay="0"
            fallbackplacement="flip"
            offset="0"
            placement="right"
            target="popoverButton-sync"
            title="Popover"
            triggers="click"
          >
            
            Hello 
            <strong>
              World!
            </strong>
          </b-popover-stub>
        </div>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Popover 
        </strong>
         
        <small>
          open/close events
        </small>
      </div>
       
      <div
        class="d-flex flex-column text-md-center"
      >
        <div
          class="p-2"
        >
          <b-btn-stub
            event="click"
            id="popoverButton-event"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="primary"
          >
            I have a popover
          </b-btn-stub>
        </div>
         
        <div
          class="p-2"
        >
          <b-btn-stub
            class="px-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Open
          </b-btn-stub>
           
          <b-btn-stub
            class="px-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Close
          </b-btn-stub>
        </div>
         
        <b-popover-stub
          boundary="scrollParent"
          boundarypadding="5"
          content=""
          delay="0"
          fallbackplacement="flip"
          offset="0"
          placement="right"
          target="popoverButton-event"
          title="Popover"
          triggers="click"
        >
          
          Hello 
          <strong>
            World!
          </strong>
        </b-popover-stub>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Popover 
        </strong>
         
        <small>
          enable/disable events
        </small>
      </div>
       
      <div
        class="d-flex flex-column text-md-center"
      >
        <div
          class="p-2"
        >
          <b-btn-stub
            event="click"
            id="popoverButton-disableevent"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="primary"
          >
            I have a popover
          </b-btn-stub>
        </div>
         
        <div
          class="p-2"
        >
          <b-btn-stub
            class="px-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Enable
          </b-btn-stub>
           
          <b-btn-stub
            class="px-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Disable
          </b-btn-stub>
        </div>
         
        <b-popover-stub
          boundary="scrollParent"
          boundarypadding="5"
          content=""
          delay="0"
          fallbackplacement="flip"
          offset="0"
          placement="right"
          target="popoverButton-disableevent"
          title="Popover"
          triggers="click"
        >
          
          Hello 
          <strong>
            World!
          </strong>
        </b-popover-stub>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Popover 
        </strong>
        <small>
          <code>
            v-b-popover
          </code>
           directive
        </small>
      </div>
       
      <b-row-stub
        tag="div"
      >
        <b-col-stub
          class="py-3 text-center"
          md="3"
          tag="div"
        >
          <b-btn-stub
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            title="Popover!"
            type="button"
            variant="primary"
          >
            Top
          </b-btn-stub>
        </b-col-stub>
         
        <b-col-stub
          class="py-3 text-center"
          md="3"
          tag="div"
        >
          <b-btn-stub
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            title="Popover!"
            type="button"
            variant="primary"
          >
            Left
          </b-btn-stub>
        </b-col-stub>
         
        <b-col-stub
          class="py-3 text-center"
          md="3"
          tag="div"
        >
          <b-btn-stub
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            title="Popover!"
            type="button"
            variant="primary"
          >
            Right
          </b-btn-stub>
        </b-col-stub>
         
        <b-col-stub
          class="py-3 text-center"
          md="3"
          tag="div"
        >
          <b-btn-stub
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            title="Popover!"
            type="button"
            variant="primary"
          >
            Bottom
          </b-btn-stub>
        </b-col-stub>
      </b-row-stub>
    </b-card-stub>
  </div>
</div>
`;
